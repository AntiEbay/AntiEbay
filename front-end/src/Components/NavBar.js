import React from "react";
import { Link } from "react-router-dom";
import Logo from "./Elements/Logo";
import "./Elements/hamburgerMenuStyles.css";
import { slide as Menu } from "react-burger-menu";

const NavBar = () => {
  return (
    <div className="flex justify-between w-full space-x-4 h-12 relative p-1 bg-slate-800">
      <div className="flex p-1.5">
        <Link to="/">
          <Logo />
        </Link>
        <div class="relative mx-auto text-gray-600 ml-2">
          <input
            class="border-2 border-gray-300 bg-white h-8 px-5 pr-16 rounded-lg text-sm focus:outline-none"
            type="search"
            name="search"
            placeholder="Search"
          />
          <button type="submit" class="absolute -right-2 bottom-0.5 mt-5 mr-4">
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-6 w-6"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div className="flex">
        <div id="sideBar" className="flex lg:hidden">
          <Menu right outerContainerId="sideBar">
            <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
              FAQ
            </button>
            <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
              Log In
            </button>
            <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
              Sign Up
            </button>
          </Menu>
        </div>
        <div className=" mx-5 hidden lg:flex">
          <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
            FAQ
          </button>
          <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
            Log In
          </button>
          <Link to="/SignUp">
            <button class="hover:bg-slate-400 cursor-pointer text-white font-bold py-2 px-2 rounded">
              Sign Up
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
};

export default NavBar;
