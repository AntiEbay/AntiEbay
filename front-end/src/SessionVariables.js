import React, { createContext, useReducer } from "react";

export const accountTypeContext = createContext();

const reducer = (state, pair) => ({ ...state, ...pair });

const initialState = {
  accountType: "0",
};

export function AccountProvider(props) {
  const [state, update] = useReducer(reducer, initialState);

  return (
    <accountTypeContext.Provider value={{ state, update }}>
      {props.children}
    </accountTypeContext.Provider>
  );
}
