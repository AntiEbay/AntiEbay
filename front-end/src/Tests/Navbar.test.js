import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../Components/NavBar";
import { AccountProvider } from "../SessionVariables";
import { BrowserRouter, Routes, Route } from "react-router-dom";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AccountProvider>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </AccountProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
