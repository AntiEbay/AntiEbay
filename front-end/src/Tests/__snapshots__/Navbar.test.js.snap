// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-between w-full space-x-4 h-12 relative p-1 bg-slate-800"
>
  <div
    className="flex p-1.5"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="flex text-sm md:text-base tracking-tighter text-white font-bold italic border-solid border-x-neutral-700"
      >
        <span
          className=" hidden md:flex"
        >
          Anti-eBay
        </span>
        <span
          className=" mt-1 flex md:hidden"
        >
          A-E
        </span>
      </div>
    </a>
    <div
      className="relative mx-auto text-gray-600 ml-2"
    >
      <input
        className="border-2 border-gray-300 bg-white h-8 px-5 pr-16 rounded-lg w-48 lg:w-96 text-sm focus:outline-none"
        label="searchQuery"
        name="searchQuery"
        onChange={[Function]}
        placeholder="Search"
        type="search"
        value=""
      />
      <a
        href="/SearchResults"
        onClick={[Function]}
      >
        <button
          className="absolute -right-2 bottom-0.5 mt-5 mr-4"
          type="submit"
        >
          <svg
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
            />
          </svg>
        </button>
      </a>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="flex lg:hidden"
      id="sideBar"
    >
      <div>
        <div
          className="bm-overlay"
          onClick={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(100%, 0, 0)",
              "MsTransform": "translate3d(100%, 0, 0)",
              "OTransform": "translate3d(100%, 0, 0)",
              "WebkitTransform": "translate3d(100%, 0, 0)",
              "background": "rgba(0, 0, 0, 0.3)",
              "height": "100%",
              "opacity": 0,
              "position": "fixed",
              "transform": "translate3d(100%, 0, 0)",
              "transition": "opacity 0.3s, transform 0s 0.3s",
              "width": "100%",
              "zIndex": 1000,
            }
          }
        />
        <div
          style={Object {}}
        >
          <div
            className="bm-burger-button"
            style={
              Object {
                "zIndex": 1000,
              }
            }
          >
            <button
              id="react-burger-menu-btn"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fontSize": 0,
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
            >
              Open Menu
            </button>
            <span>
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "0%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "40%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "80%",
                  }
                }
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="bm-menu-wrap"
          id=""
          style={
            Object {
              "MozTransform": "translate3d(100%, 0, 0)",
              "MsTransform": "translate3d(100%, 0, 0)",
              "OTransform": "translate3d(100%, 0, 0)",
              "WebkitTransform": "translate3d(100%, 0, 0)",
              "height": "100%",
              "position": "fixed",
              "right": 0,
              "transform": "translate3d(100%, 0, 0)",
              "transition": "all 0.5s",
              "width": "300px",
              "zIndex": 1100,
            }
          }
        >
          <div
            className="bm-menu"
            style={
              Object {
                "boxSizing": "border-box",
                "height": "100%",
                "overflow": "auto",
              }
            }
          >
            <nav
              className="bm-item-list"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <button
                className="bm-item hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                FAQ
              </button>
              <a
                className="bm-item"
                href="/SignIn"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <button
                  className="hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
                >
                  Log In
                </button>
              </a>
              <a
                className="bm-item"
                href="/SignUp"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <button
                  className="hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
                >
                  Sign Up
                </button>
              </a>
            </nav>
          </div>
          <div
            style={Object {}}
          >
            <div
              className="bm-cross-button"
              style={
                Object {
                  "height": 24,
                  "position": "absolute",
                  "right": 8,
                  "top": 8,
                  "width": 24,
                }
              }
            >
              <button
                id="react-burger-cross-btn"
                onClick={[Function]}
                style={
                  Object {
                    "background": "transparent",
                    "border": "none",
                    "cursor": "pointer",
                    "fontSize": 0,
                    "height": "100%",
                    "left": 0,
                    "margin": 0,
                    "padding": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={-1}
              >
                Close Menu
              </button>
              <span
                style={
                  Object {
                    "position": "absolute",
                    "right": "14px",
                    "top": "6px",
                  }
                }
              >
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(45deg)",
                      "width": 3,
                    }
                  }
                />
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(-45deg)",
                      "width": 3,
                    }
                  }
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className=" mx-5 hidden lg:flex"
    >
      <button
        className="hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
      >
        FAQ
      </button>
      <a
        href="/SignIn"
        onClick={[Function]}
      >
        <button
          className="hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
        >
          Log In
        </button>
      </a>
      <a
        href="/SignUp"
        onClick={[Function]}
      >
        <button
          className="hover:bg-sky-700 cursor-pointer text-white font-bold py-2 px-2 rounded"
        >
          Sign Up
        </button>
      </a>
    </div>
  </div>
</div>
`;
